<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Partitioning and aggregating cross-tissue and 
    tissue-specific genetic effects in identifying gene-trait associations.">
<title>Multi-tissue Transcriptome-Wide Association Studies ‚Ä¢ MTWAS</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Multi-tissue Transcriptome-Wide Association Studies">
<meta property="og:description" content="Partitioning and aggregating cross-tissue and 
    tissue-specific genetic effects in identifying gene-trait associations.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">MTWAS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="articles/mtwas.html">Tutorial</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/szcf-weiya/MTWAS/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="mtwas-multi-tissue-transcriptome-wide-association-studies">MTWAS: Multi-tissue Transcriptome-Wide Association Studies<a class="anchor" aria-label="anchor" href="#mtwas-multi-tissue-transcriptome-wide-association-studies"></a>
</h1></div>
<p>MTWAS is an R package for the paper</p>
<blockquote>
<p>Song, S., Wang, L., Hou, L., &amp; Liu, J. S. (2023). Partitioning and aggregating cross-tissue and tissue-specific genetic effects in identifying gene-trait associations.</p>
</blockquote>
<div class="section level2">
<h2 id="white_check_mark-prerequisites">
<span class="emoji" data-emoji="white_check_mark">‚úÖ</span> Prerequisites<a class="anchor" aria-label="anchor" href="#white_check_mark-prerequisites"></a>
</h2>
<p>The software is developed and tested in Linux and Windows environments.</p>
<ul>
<li>R (&gt;=3.6)</li>
<li>GNU Scientific Library (GSL) (&gt;=2.3)</li>
</ul>
</div>
<div class="section level2">
<h2 id="hammer_and_wrench-installation">
<span class="emoji" data-emoji="hammer_and_wrench">üõ†Ô∏è</span> Installation<a class="anchor" aria-label="anchor" href="#hammer_and_wrench-installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"szcf-weiya/MTWAS"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="rocket-getting-started">
<span class="emoji" data-emoji="rocket">üöÄ</span> Getting Started<a class="anchor" aria-label="anchor" href="#rocket-getting-started"></a>
</h2>
<p><span class="emoji" data-emoji="smiley">üòÉ</span> You could either download the pre-trained GTEx v8 MTWAS weights, or generate your own files!</p>
</div>
<div class="section level2">
<h2 id="one-run-mtwas-with-pre-trained-gtex-v8-weights-stareasy-and-recommended">
<span class="emoji" data-emoji="one">1Ô∏è‚É£</span> Run MTWAS with pre-trained GTEx v8 weights (<span class="emoji" data-emoji="star">‚≠ê</span>Easy and Recommended)<a class="anchor" aria-label="anchor" href="#one-run-mtwas-with-pre-trained-gtex-v8-weights-stareasy-and-recommended"></a>
</h2>
<div class="section level3">
<h3 id="download-the-pre-trained-files">Download the pre-trained files:<a class="anchor" aria-label="anchor" href="#download-the-pre-trained-files"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> gtex_v8_mtwas_eqtls.tar.gz https://cloud.tsinghua.edu.cn/f/5633911d7c39431b8be8/<span class="pp">?</span>dl=1 <span class="at">--no-check-certificate</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-zxvf</span> gtex_v8_mtwas_eqtls.tar.gz</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prepare-the-gwas-summary-statistics-in-the-following-format-including-the-header-line">Prepare the GWAS summary statistics in the following format (including the header line):<a class="anchor" aria-label="anchor" href="#prepare-the-gwas-summary-statistics-in-the-following-format-including-the-header-line"></a>
</h3>
<pre><code>   chr        rsid      a1    a2       z         
    1      rs4040617    G     A     -0.199    
    1      rs4075116    C     T      0.646     
    1      rs9442385    T     G     -0.016    
    ...</code></pre>
<p><strong>chr</strong>: chromosome</p>
<p><strong>rsid</strong>: SNP rsid</p>
<p><strong>a1</strong>: reference allele</p>
<p><strong>a2</strong>: alternative allele</p>
<p><strong>z</strong>: GWAS z score</p>
</div>
<div class="section level3">
<h3 id="mtwas-analysis">MTWAS analysis:<a class="anchor" aria-label="anchor" href="#mtwas-analysis"></a>
</h3>
<p>We use chromosome 22 on whole blood as an example. The list of cell types is detailed in <code>ct_use.RData</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/szcf-weiya/MTWAS" class="external-link">MTWAS</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"summary_stats"</span><span class="op">)</span> <span class="co">## EXAMPLE GWAS summary stats (could be specified by users, format: a data.frame with colnames: chr, rsid, a1, a2, z)</span></span>
<span><span class="va">chr</span> <span class="op">=</span> <span class="fl">22</span></span>
<span><span class="va">cell_type</span> <span class="op">=</span> <span class="st">'Whole_Blood'</span></span>
<span><span class="co">### remember to change the path to the downloaded folder!!!</span></span>
<span><span class="co">## load twas bim files (downloaded)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'./gtex_v8_mtwas_eqtls/twas_bim_chr'</span>, <span class="va">chr</span>, <span class="st">'.RData'</span><span class="op">)</span><span class="op">)</span>  </span>
<span><span class="co">## load twas eqtl files (downloaded)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'./gtex_v8_mtwas_eqtls/'</span>, <span class="va">cell_type</span>, <span class="st">'/twas_eqtl_chr'</span>, <span class="va">chr</span>, <span class="st">'.RData'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## Run mtwas and derive the gene-trait association test statistics</span></span>
<span><span class="va">results</span> <span class="op">=</span> <span class="fu"><a href="reference/run_mtwas_easy.html">run_mtwas_easy</a></span><span class="op">(</span><span class="va">summary_stats</span>, <span class="va">twas_bim</span>, <span class="va">twas_eqtl</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="two-train-your-own-weights">
<span class="emoji" data-emoji="two">2Ô∏è‚É£</span> Train your own weights<a class="anchor" aria-label="anchor" href="#two-train-your-own-weights"></a>
</h2>
<p>We also provide functions to train MTWAS with your own datasets!</p>
<div class="section level3">
<h3 id="data-preparation">Data preparation<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h3>
<p>In order to derive your own MTWAS weights, three types of data are necessary. There is a demo dataset built in our R package:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/szcf-weiya/MTWAS" class="external-link">MTWAS</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'demo'</span><span class="op">)</span></span>
<span><span class="co"># demo$dat</span></span>
<span><span class="co"># demo$E.info</span></span>
<span><span class="co"># demo$E</span></span></code></pre></div>
<div class="section level4">
<h4 id="id_1-genotype-files-dat">(1) Genotype files (dat)<a class="anchor" aria-label="anchor" href="#id_1-genotype-files-dat"></a>
</h4>
<p>Format: a <strong>list</strong> of plink bfiles, including bim, fam, bed</p>
<p>One could use the R function <code>read_plink</code> in package <code>EBPRS</code> to read the plink files:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EBPRS</span><span class="op">)</span></span>
<span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">read_plink</span><span class="op">(</span><span class="st">'PATH_TO_PLINK_BFILE'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_2-gene-information-einfo">(2) Gene information (E.info)<a class="anchor" aria-label="anchor" href="#id_2-gene-information-einfo"></a>
</h4>
<p>Genes that we are interested in to derive the gene-trait associations.</p>
<p>Format: a <strong>data.frame</strong> in the following format (including the header line):</p>
<pre><code>   chr        start        end        gene         
    22      15528192    15529139     OR11H1   
    22      15690026    15721631     POTEH          
    ...</code></pre>
</div>
<div class="section level4">
<h4 id="id_3-gene-expression-data-e">(3) Gene expression data (E)<a class="anchor" aria-label="anchor" href="#id_3-gene-expression-data-e"></a>
</h4>
<p>A <strong>list</strong> including all the <strong>gene expression data</strong>, the length of the list is the number of tissues. Each element is a sample*gene matrix.</p>
<p>Each matrix should have <strong>rownames</strong> (overlapped with <code>dat$fam$V2</code>), and <strong>colnames</strong> (overlapped with <code>E.info$gene</code>)</p>
<p>The orders of the columns and rows of the matrices are not necessary to be the same. The matrices can have NAs.</p>
<p><span class="emoji" data-emoji="exclamation">‚ùó</span> Please NOTE that the position of <code>dat$bim$V4</code> and <code>E.info</code> should be the same build (e.g., both are hg19, or hg38, or etc.)</p>
</div>
</div>
<div class="section level3">
<h3 id="data-imputation-and-formatting">Data imputation and formatting<a class="anchor" aria-label="anchor" href="#data-imputation-and-formatting"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/szcf-weiya/MTWAS" class="external-link">MTWAS</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'demo'</span><span class="op">)</span></span>
<span><span class="co">### substitute the input with your own dataset</span></span>
<span><span class="va">twas_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/format_twas_dat.html">format_twas_dat</a></span><span class="op">(</span>E<span class="op">=</span><span class="va">demo</span><span class="op">$</span><span class="va">E</span>, E.info<span class="op">=</span><span class="va">demo</span><span class="op">$</span><span class="va">E.info</span>, dat<span class="op">=</span><span class="va">demo</span><span class="op">$</span><span class="va">dat</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">twas_dat</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="model-training">Model training<a class="anchor" aria-label="anchor" href="#model-training"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load TWAS data</span></span>
<span><span class="co"># select cross-tissue eQTLs</span></span>
<span><span class="va">ct.eQTL</span> <span class="op">=</span> <span class="fu"><a href="reference/select.ct.eQTL.html">select.ct.eQTL</a></span><span class="op">(</span><span class="va">twas_dat</span>, verbose <span class="op">=</span> <span class="cn">F</span>, ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co"># select tissue-specific eQTLs</span></span>
<span><span class="va">list.eQTL</span> <span class="op">=</span> <span class="fu"><a href="reference/select.ts.eQTL.html">select.ts.eQTL</a></span><span class="op">(</span><span class="va">twas_dat</span>, ct.eQTL <span class="op">=</span> <span class="va">ct.eQTL</span>, ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extract-cross-tissue-eqtls">Extract cross-tissue eQTLs<a class="anchor" aria-label="anchor" href="#extract-cross-tissue-eqtls"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_name</span> <span class="op">=</span> <span class="st">'IL17RA'</span> <span class="co">## gene name</span></span>
<span><span class="va">gene_index</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ct.eQTL</span><span class="op">)</span><span class="op">==</span><span class="va">gene_name</span><span class="op">)</span></span>
<span><span class="co">## ct-eQTLs for gene IL17RA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">list.eQTL</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="va">gene_index</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">`common.snp`</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extract-tissue-specific-eqtls">Extract tissue-specific eQTLs<a class="anchor" aria-label="anchor" href="#extract-tissue-specific-eqtls"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_name</span> <span class="op">=</span> <span class="st">'CCT8L2'</span> <span class="co">## gene name</span></span>
<span><span class="va">gene_index</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ct.eQTL</span><span class="op">)</span><span class="op">==</span><span class="va">gene_name</span><span class="op">)</span></span>
<span><span class="va">tissue_index</span> <span class="op">=</span> <span class="fl">1</span> <span class="co">## tissue specific</span></span>
<span><span class="co">## ts-eQTLs for gene CCT8L2 on tissue 1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">list.eQTL</span><span class="op">[[</span><span class="va">tissue_index</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="va">gene_index</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">`single.snp`</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="gene-trait-association-tests">Gene-trait association tests<a class="anchor" aria-label="anchor" href="#gene-trait-association-tests"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load GWAS summary statistics (data.frame, colnames: rsid, a1, a2, chr, z)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"summary_stats"</span><span class="op">)</span></span>
<span><span class="co"># association test</span></span>
<span><span class="fu"><a href="reference/twas.single.trait.html">twas.single.trait</a></span><span class="op">(</span><span class="va">summary_stats</span>, <span class="va">twas_dat</span>, <span class="va">list.eQTL</span><span class="op">)</span></span></code></pre></div>
<p>The details of output and data formats can be found in the auto-generated vignette: <a href="https://hohoweiya.xyz/MTWAS/articles/mtwas.html" class="uri">https://hohoweiya.xyz/MTWAS/articles/mtwas.html</a></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/szcf-weiya/MTWAS/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/szcf-weiya/MTWAS/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing MTWAS</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://hohoweiya.xyz" class="external-link">Lijun Wang</a> <br><small class="roles"> Author, maintainer </small>  </li>
<li>
<a href="https://github.com/shuangsong0110" class="external-link">Shuang Song</a> <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://hohoweiya.xyz" class="external-link">Lijun Wang</a>, <a href="https://github.com/shuangsong0110" class="external-link">Shuang Song</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
